test/build-combination/HeaderSelfContain/headerSelfContain.cmake

```diff
@@ -0,0 +1,45 @@
+set( COMPILE_OPTIONS
```

Comment:
target_compile_options could be used here.

Reply:
Updated